brickstrap.sh

```diff
@@ -19,29 +20,51 @@
 # http://twitter.com/kvz
 #
 
-### Configuration
+### Runtime
 #####################################################################
+set -e
+
+# Bash will remember & return the highest exitcode in a chain of pipes.
+set -o pipefail
 
 # Environment variables
 [ -z "${LOG_LEVEL}" ] && LOG_LEVEL="3" # 4 = debug -> 0 = fail
 
-# Commandline options. This defines the usage page
+#
+# Get the brickstrap source directory
+#
+function br_script_path()
+{
+    SCRIPT_PATH=$(readlink -f "$0")
+    SCRIPT_PATH=$(dirname "$SCRIPT_PATH")
+    echo -n "$SCRIPT_PATH"
+}
 
-read -r -d '' usage <<-'EOF'
-  usage: brickstrap [-b <board>] [-d <dir>] <command>
+. "$(br_script_path)/brickstrap-components.sh"
+
+# Commandline options. This defines the usage page
+BRP_USAGE=$(cat <<'EOF'
+Usage: brickstrap -p <project> [-c <component>] [-d <dir>] <command>
```

Comment:
Let's leave these kind of behaviour changes out of the scope of this PR. There's a lot of scope for bugs and unintended consequences:

 * what are the semantics of the -f flag? 
 * what if you do want a new config? 
 * how do you know all settings are preserved and not just the ones which are hit by the code path of the current command? 
 * how do you avoid the 'history' simply growing and growing endlessly, leading to worse and worse start up time and needless waste of disk space?

Perhaps you could setup an issue for exploring this/tracking these changes?